### ISSUE TITLE
BBC programme downloaded from dash stream converted rather than remuxed by ffmpeg

### ISSUE THREAD

**Issue by jefro108** [Ascoware/get-iplayer-automator#256]
When I tried to download https://www.bbc.co.uk/iplayer/episode/p07hbyjc/untouchable-the-rise-and-fall-of-harvey-weinstein it ended up being downloaded from a dash stream and produced a file 'Untouchable The Rise and Fall of Harvey Weinstein.dash.ts' which ffmpeg then proceded to convert rather than remux!

`INFO: Command: "/Applications/Get iPlayer Automator.app/Contents/MacOS/ffmpeg" "-loglevel" "fatal" "-stats" "-y" "-i" "/Volumes/media/Downloads/TV Shows/Untouchable The Rise and Fall of Harvey Weinstein/Untouchable The Rise and Fall of Harvey Weinstein.dash.ts" "-c:v" "libx264" "-c:a" "copy" "-preset" "ultrafast" "-movflags" "+faststart" "-bsf:a" "aac_adtstoasc" "/Volumes/media/Downloads/TV Shows/Untouchable The Rise and Fall of Harvey Weinstein/Untouchable The Rise and Fall of Harvey Weinstein.partial.mp4"`

Since the file already contains a H264 and MacOS will even play the file with QuickLook this doesn't seem necessary - plus it takes ages!

```
File
Name: Untouchable The Rise and Fall of Harvey Weinstein.dash.ts
Size: 2.92 GB (2922224248 bytes)
Kind: Gen2Letter02
UTI: public.mpeg-2-transport-stream
Location: /Volumes/media/Downloads/TV Shows/Untouchable The Rise and Fall of Harvey Weinstein
Created: 1 January 1904 at 00:00:00
Modified: 4 September 2019 at 03:20:32
Container
ID: 1 (0x1)
Format: MPEG-TS
Duration: 1 h 34 min 8 s 640 ms
Overall bit rate mode: Variable
Overall bit rate: 4 138 kb/s
Video
ID: 257 (0x101)
Menu ID: 1 (0x1)
Format: AVC
Format/Info: Advanced Video Codec
Format profile: High@L3.2
Format settings: CABAC / 4 Ref Frames
Format settings, CABAC: Yes
Format settings, ReFrames: 4 frames
Codec ID: 27
Duration: 1 h 34 min 10 s 760 ms
Width: 1 280 pixels
Height: 720 pixels
Pixel aspect ratio: 1.000
Display aspect ratio: 16:9
Frame rate: 50.000 FPS
Frame count: 282538
Color space: YUV
Chroma subsampling: 4:2:0
Bit depth: 8 bits
Scan type: Progressive
Color range: Limited
Color primaries: BT.709
Transfer characteristics: BT.709
Matrix coefficients: BT.709
Audio
ID: 256 (0x100)
Menu ID: 1 (0x1)
Format: AAC LC
Format/Info: Advanced Audio Codec Low Complexity
Format version: Version 4
Muxing mode: ADTS
Codec ID: 15-2
Duration: 1 h 34 min 10 s 688 ms
Bit rate mode: Variable
Channel(s): 2 channels
Channel layout: L R
Sampling rate: 48.0 kHz
Frame rate: 46.875 FPS (1024 SPF)
Compression mode: Lossy
Language: English
Menu/Chapters
ID: 4096 (0x1000)
Menu ID: 1 (0x1)
Duration: 1 h 34 min 8 s 640 ms
List: 256 (0x100) (AAC, English) / 257 (0x101) (AVC)
Language: English
Service name: Service01
Service provider: FFmpeg
Service type: digital television
```

Surely a simple `-c:v copy` would suffice?

**Comment by skovatch** [comment]
What quality settings are you using? I just started a download with BBC TV "Better" and "Best" I'm not getting a DASH stream. I can try to investigate more but I can't reproduce it right now.

GiA used to have logic that detected DASH and tried to convert it with ffmpeg but I removed it when get_iplayer started doing it for you.

**Comment by jefro108** [comment]
Even though I am in the UK I use a VPN on my laptop by default to keep my connection secure. Sometimes it has trouble connecting with the BBC without getting 403 errors and ends up using a dash stream which then proceeded to be converted by ffmpeg!

**Comment by omensan** [comment]
The DASH stream downloads M4A and M4V, GiA invokes FFmpeg to convert to MPEG-TS with "-c:v copy", downloads (and often fails to download) subtitles, then re-invokes FFmpeg with "c:v libx264" and re-encodes.

Is this an issue triggered by the subtitle download failure somehow?  I would think the default behaviour would be to "-c:v copy", and re-encode only if that attempt exits in failure state.

If not the default, I would propose this should be an option in the <Advanced> Preference tab.  Not everyone expects a perfectly formed MP4 that is fully iTunes compliant; VLC, MPV and others will play the original M4V/M4A, the muxed MPEG-TS, and a manually remuxed MP4 just fine.

**Comment by skovatch** [comment]
I have noticed this now as well, unfortunately. All of my 'smart DNS' providers are generating 403 errors, but eventually one works and provides a DASH stream. Then it does the full conversion which kicks the processor into high gear. I can try changing the logic in GiA's local copy of get_iplayer to just do a copy when we get a DASH stream and see how it works.

**Comment by skovatch** [comment]
OK, removing the aac_adtstoasc still produces a usable MP4 that can be played in TV.app even with a DASH download. According to the get_iplayer forums the 403 errors are happening due to a change on the BBC side and will be addressed in a get_iplayer update. Once that happens I can re-evaluate the situation and see if removing the conversion is still needed.

**Comment by jefro108** [comment]
How about removing the libx264 as well ie changing "-c:v" "libx264" to "-c:v" "copy"? 

Since the .ts file plays in QuickLook fine - so it is pretty Apple compatible before conversion! 😊

```
INFO: Converting to MP4 and embedding subtitles
INFO: Command: "/Applications/Get iPlayer Automator.app/Contents/MacOS/ffmpeg" "-loglevel" "fatal" "-stats" "-y" "-i" "/Volumes/media/Downloads/TV Shows/Clangers/Clangers.s03e18.Junk Planet.dash.ts" "-i" "/Volumes/media/Downloads/TV Shows/Clangers/Clangers.s03e18.Junk Planet.srt" "-c:v" "libx264" "-c:a" "copy" "-preset" "ultrafast" "-movflags" "+faststart" "-c:s" "mov_text" "-metadata:s:s:0" "language=eng" "-bsf:a" "aac_adtstoasc" "/Volumes/media/Downloads/TV Shows/Clangers/Clangers.s03e18.Junk Planet.partial.mp4"
```

**Comment by skovatch** [comment]
I'm restoring back to what the base get_iplayer does, so none of that will happen now. Expect a new release soon.

**Comment by skovatch** [comment]
Fixed in 1.17
